### ISSUE TITLE
[Performance] Extremely slow inference latency on A100-80GB (~9s per image) without Flash Attention

### ISSUE THREAD

**Issue by usavich20** [QwenLM/Qwen3-VL-Embedding#33]
#### Description
I am experiencing significant performance issues when using the official script for `Qwen3-VL-Embedding-8B`. On an NVIDIA A100-80GB GPU, generating an embedding for a single 512x512 image takes approximately **9 seconds**. 

Profiling indicates that the **Model Forward Pass** is the primary bottleneck, accounting for over 98% of the total execution time.

#### Environment
*   **Hardware:** NVIDIA A100-SXM4-80GB
*   **CUDA:** 12.8
*   **PyTorch:** 2.9.0+cu128
*   **Transformers:** 4.57.3
*   **Constraint:** ⚠️ `flash-attention` is **NOT** installed due to strict enterprise environment restrictions.

#### Performance Metrics
*   **Input:** 512×512 image (approx. 262K pixels, pre-loaded via PIL)
*   **Precision:** `torch.bfloat16`
*   **Average Latency:** ~9164 ms

**Detailed Breakdown:**
| Component | Latency (ms) | Percentage |
| :--- | :--- | :--- |
| `format_model_input` | 0.00 ms | 0% |
| `process_vision_info` | 0.62 ms | <0.1% |
| `processor` (tokenization) | 96.24 ms | ~1% |
| **Model forward pass** | **9201.36 ms** | **~98%** |
| `Pooling + Normalization` | 0.19 ms | <0.1% |

#### Reproduction Code
```python
import torch
import time
from PIL import Image
from src.models.qwen3_vl_embedding import Qwen3VLEmbedder

# Initialize model
model = Qwen3VLEmbedder(
    model_name_or_path="Qwen/Qwen3-VL-Embedding-8B",
    dtype=torch.bfloat16
)

img = Image.new('RGB', (512, 512), color='red')

# Warm-up
_ = model.process([{"image": img}])

# Benchmark
for i in range(3):
    start = time.time()
    embeddings = model.process([{"image": img}])
    print(f"Run {i+1}: {(time.time()-start)*1000:.2f} ms")

# Output: ~9000 ms per iteration
```

#### Questions
1. Is this level of latency expected when `flash-attention` is unavailable?
2. Are there any recommended optimizations for the Vision Encoder that do not require external CUDA extensions like `flash-attn`? 
3. Or should I wait for official support in future releases (e.g., vLLM 0.14.0) for better performance?

Thank you for your help!
---

#### 问题描述
在使用官方脚本运行 `Qwen3-VL-Embedding-8B` 时，推理速度极慢。在 NVIDIA A100-80GB 环境下，处理单张 512x512 图片的 embedding 需要约 **9 秒**。

性能分析显示，**模型前向传播 (Model Forward Pass)** 是核心瓶颈，占总耗时的 98% 以上。

#### 环境信息
*   **硬件:** NVIDIA A100-SXM4-80GB
*   **CUDA:** 12.8
*   **PyTorch:** 2.9.0+cu128
*   **Transformers:** 4.57.3
*   **限制条件:** ⚠️ 由于企业环境受限，**无法安装** `flash-attention`。

#### 性能数据
*   **配置:** 512×512 图片 (约 262K 像素，PIL 预加载)
*   **精度:** `torch.bfloat16`
*   **平均耗时:** ~9164 ms

**耗时分布详情:**
| 步骤 | 耗时 (ms) | 占比 |
| :--- | :--- | :--- |
| `format_model_input` | 0.00 ms | 0% |
| `process_vision_info` | 0.62 ms | <0.1% |
| `processor` (分词/预处理) | 96.24 ms | ~1% |
| **Model forward pass (模型前向)** | **9201.36 ms** | **~98%** |
| `Pooling + Normalization` | 0.19 ms | <0.1% |

#### 复现代码
（代码同上，此处略）

#### 待确认问题
1. 在没有 Flash Attention 的情况下，这种性能表现是否符合预期？
2. 针对无法安装 `flash-attn` 的受限环境，是否有其他优化 Vision Encoder 的建议？
3. 还是说我应该等待后续版本的官方支持（例如 vLLM 0.14.0）以获得更好的性能？

感谢！

**Comment by Catstyle** [comment]
The same issue here.

Run VL-embedding 2B on a 4090 with FA2, need almost 9 minutes for 8 images in a batch

**Comment by 8igfive** [comment]
We also encountered very slow image/video encoding performance when using **PyTorch 2.9**. In our case, the issue was resolved by downgrading PyTorch to **2.8**, which is why you can see `"torch==2.8.*"` specified in the `pyproject.yaml`.

**Comment by Matagi1996** [comment]
Can confirm that downgrading to 2.8 solved the issue.
As for the reason why, the Image Encoder uses conv3d and torch2.9 seems to have an issue with it currently.
https://github.com/pytorch/pytorch/issues/168167

**Comment by michaelsieghartsleitner** [comment]
Thank you!
Do you have any benchmarks or data on throughput, latency, or overall performance when using PyTorch 2.8 on an A100-80GB @usavich20 ?

**Comment by weathon** [comment]
I downgraded to 2.8 and it is much better but still slow

**Comment by usavich20** [comment]
> I downgraded to 2.8 and it is much better but still slow

You can try limiting the number of pixels in each image.In my case it feels acceptable with {"max_pixels": 262144}
